src/sources/syslog.rs

```diff
@@ -200,16 +200,22 @@ impl TcpSource for SyslogTcpSource {
 #[derive(Clone, Debug)]
 struct SyslogDecoder {
     other: LinesCodec,
+    is_discarding: bool,
+    discard_chars: Option<usize>,
```

Comment:
`discard_chars` is only examined when `is_discarding` is `true`. This strongly suggests it's state data that may be better encapsulated in a proper state, like this (but probably with better variant names):
```rust
enum State {
    NotDiscarding,
    Discarding(usize),
    DiscardingToEOL,
}
```
I also wonder if `in_octet` could be rolled into this state as well, but I can't quite tell from the logic here.

Reply:
Yes, that's much better.